fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -1026,6 +1027,35 @@ private FragmentExecParams getMaxParallelismScanFragmentExecParams() {
         return maxParallelismExecParams;
     }
 
+    private Set<PlanNodeId> getExchangeNodes(PlanNode node) {
```

Comment:
Why you need to change the code in Coordinator? which is bloodcurdling.

Reply:
For an PlanFragment whose left-deepmost child is ExchangeNode, its child PlanFragment that has the maximum numInstances * pipelineDop is taken into consideration, if the child PlanFragment's DataSink is partitioned, then pipeline paralellization is adopted, otherwise fragment instance parallelization is adopted